homeassistant/helpers/data_entry_flow.py

```diff
@@ -60,7 +61,8 @@ async def post(self, request, data):
 
         try:
             result = await self._flow_mgr.async_init(
-                handler, context={"source": config_entries.SOURCE_USER}
+                handler,  # type: ignore
```

Comment:
I would like to draw attention here -- the type ignore is required because this function accepts a list in `data["handler"]` in addition to a str, and converts it to a tuple. But I didn't find evidence that anything but a str would actually be supported by the underlying code, so a proper fix could be to remove the list/typle mangling code altogether.

Reply:
On the other hand I would prefer if merging this PR would not be delayed by this, and this fixed in a separate one unless someone can say straight away that the above (or a similar simple fix) should be done here-

Reply:
Dug though 534aa0e4b54b992ac55de0ae576276e94089ce49 and couldn't figure out why a `list` is allowed.  Likely Paulus needs answer that one.

There is a conflict that needs to be resolved as well

Reply:
Conflict resolved again. Would be great to have this merged with or without the above addressed, to avoid chasing conflicts.